pages/[id].js

```diff
@@ -45,28 +55,70 @@ export default function PlantDetails({ plants, onDeletePlant, onEditPlant }) {
     onEditPlant(plant.id, updatedPlant);
     setShowEdit(false);
   }
+  function toggleUploadImage() {
+    setUploadOpen(!uploadOpen);
+  }
 
-  const router = useRouter();
-  const { id } = router.query;
+  async function handleImageUpload(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
 
-  if (!router.isReady) return null;
+    try {
+      const response = await fetch("/api/upload", {
+        method: "POST",
+        body: formData,
+      });
 
-  const plant = plants.find((plant) => plant.id === id);
+      if (response.ok) {
+        const data = await response.json();
+        const newImageUrl = data.secure_url;
 
-  if (!plant) return <p>Plant not found</p>;
+        const updatedPlant = { ...plant, imageUrl: newImageUrl };
+
+        onEditPlant(plant.id, updatedPlant);
+        setUploadOpen(false);
+      } else {
+        console.error("Failed to upload image.");
```

Comment:
Consider to provide user feedback on failed image upload.An alert could be enough for the first step. 

```suggestion
      } else {
        console.error("Failed to upload image.");
         alert("Image upload failed. Please try again.");
```